Common/Extensions.cs

```diff
@@ -2338,5 +2449,38 @@ public static OptionChainUniverse CreateOptionChain(this IAlgorithm algorithm, S
 
             return new OptionChainUniverse(optionChain, settings, algorithm.LiveMode);
         }
+
+        /// <summary>
+        /// Inverts the specified <paramref name="right"/>
+        /// </summary>
+        public static OptionRight Invert(this OptionRight right)
+        {
+            switch (right)
+            {
+                case OptionRight.Call: return OptionRight.Put;
+                case OptionRight.Put:  return OptionRight.Call;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(right), right, null);
+            }
+        }
+
+        public static void Invoke<T>(this T value, params Action<T>[] invokers)
+        {
+            var type = typeof(T);
+            if (!type.IsEnum)
+            {
+                throw new ArgumentException($"Invoke<T> is only for enum types. {type.Name} is not supported.");
+            }
+
+            var values = type.GetEnumValues();
+            for (var i = 0; i < values.Length; i++)
+            {
+                if (Equals(value, values.GetValue(i)))
+                {
+                    invokers[i](value);
+                    return;
```

Comment:
I think this got removed... I can't seem to find it anymore. Ahh, I remember what it did -- based on an enumeration value it would invoke a delegate. It's basically a sexier switch statement 👅  -- shame I can't see what the whole function was -- I recall it being kinda cool though.